src/main/java/com/insightfullogic/honest_profiler/core/profiles/lean/LeanNode.java

```diff
@@ -0,0 +1,287 @@
+    {
+        return false;
+    }
+
+    // Aggregation Methods
+
+    /**
+     * Aggregates teh information from a child {@link StackFrame} into the children of this LeanNode, updating the total
+     * data for this LeanNode in the process.
+     *
+     * The add method is called from the LogCollector, on the parent with the next (potentially last) child from the
+     * trace.
+     *
+     * @param nanos the number of nanoseconds between the {@link TraceStart} preceding the stack trace sample and the
+     *            {@link TraceStart} following it
+     * @param child the child {@link FrameInfo} of the current LeanNode
+     * @param last a boolean indicating if the child is the last in the stack trace sample
+     * @return this node
+     */
+    public LeanNode add(long nanos, FrameInfo child, boolean last)
+    {
+        // Non-self add, which updates total time and sample count only.
+        data.add(nanos, false);
+
+        return childMap.compute(
+            child,
+            (k, v) ->
+            // Create a new LeanNode if no children have been recorded for this FrameInfo
+            v == null ? (last ?
+            // New child is the last in stack, use "Self constructor".
+                new LeanNode(child, nanos, this)
+                // New child is not the last in stack, use "Non-self constructor".
+                : new LeanNode(child, this))
+                // Aggregate the information into an existing child
+                : (last ?
+                // Child is last in stack, use the "self add" on existing child
+                    v.addSelf(nanos) :
+                    // Child is not last in stack, return existing child (its "total" numbers will be updated when its
+                    // child is added)
+                    v));
+    }
+
+    /**
+     * Aggregate the self and total data in the {@link NumericInfo} for this LeanNode.
+     *
+     * @param nanos the self time for the frame
+     * @return this object
+     */
+    private LeanNode addSelf(long nanos)
+    {
+        data.add(nanos, true);
+        return this;
+    }
+
+    // Tree-related Methods
+
+    /**
+     * Returns a {@link Stream} containing this LeanNode and all its descendants.
+     *
+     * @return a {@link Stream} containing this LeanNode and all its descendants
+     */
+    public Stream<LeanNode> flatten()
+    {
+        return concat(of(this), childMap.values().stream().flatMap(LeanNode::flatten));
+    }
+
+    // Copy Methods
+
+    /**
+     * Copy method for the top of the tree, which has no parent.
+     *
+     * @return a copy of this object
+     */
+    public LeanNode copy()
+    {
+        return new LeanNode(this, null);
+    }
+
+    /**
+     * Copy method for children.
+     *
+     * @param newParent the new parent for the children
+     * @return a copy of this object
+     */
+    public LeanNode copy(LeanNode newParent)
+    {
+        return new LeanNode(this, newParent);
+    }
```

Comment:
Fixed in upcoming commit, by getting rid of the two copy() methods. One was unused, and the one referenced here was used only in a single location where it didn't hurt readability to replace it by a constructor call.